eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/ExplorerApi.scala

```diff
@@ -130,60 +132,58 @@ object ExplorerApi {
+          case Left(otherError) => throw otherError
+          case Right(json) if json \ "error" == JString(s"Block $blockCount not found.") => Seq.empty
+          case Right(block) =>
+            val JInt(height) = block \ "height"
+            val JInt(version) = block \ "ver"
+            val JString(time) = block \ "time"
+            val JInt(bits) = block \ "bits"
+            val JInt(nonce) = block \ "nonce"
+            val previousBlockHash = (block \ "prev_block").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
+            val merkleRoot = (block \ "mrkl_root").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
+            val header = BlockHeader(version.toLong, previousBlockHash, merkleRoot, OffsetDateTime.parse(time).toEpochSecond, bits.toLong, nonce.toLong)
+            Seq(BlockHeaderAt(BlockHeight(height.toLong), header))
         })
-        .map(blocks => blocks.map(block => {
-          val JInt(height) = block \ "height"
-          val JInt(version) = block \ "ver"
-          val JString(time) = block \ "time"
-          val JInt(bits) = block \ "bits"
-          val JInt(nonce) = block \ "nonce"
-          val previousBlockHash = (block \ "prev_block").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
-          val merkleRoot = (block \ "mrkl_root").extractOpt[String].map(ByteVector32.fromValidHex(_).reverse).getOrElse(ByteVector32.Zeroes)
-          val header = BlockHeader(version.toLong, previousBlockHash, merkleRoot, OffsetDateTime.parse(time).toEpochSecond, bits.toLong, nonce.toLong)
-          BlockHeaderAt(BlockHeight(height.toLong), header)
-        }))
     } yield header
   }
 
   /** Explorer API based on Esplora: see https://github.com/Blockstream/esplora/blob/master/API.md. */
   sealed trait Esplora extends Explorer with SupportsTor {
-    implicit val sb: SttpBackend[Future, Nothing]
+    implicit val sb: SttpBackend[Future, capabilities.WebSockets]
 
     override def getLatestHeaders(baseUri: Uri, currentBlockHeight: BlockHeight)(implicit context: ActorContext[Command]): Future[LatestHeaders] = {
       implicit val ec: ExecutionContext = context.system.executionContext
       for {
-        headers <- sttp.readTimeout(10 seconds).get(baseUri.path(baseUri.path :+ "blocks"))
+        headers <- basicRequest.readTimeout(10 seconds).get(baseUri.addPath("blocks"))
           .response(asJson[JArray])
-          .send()
+          .send(sb)
           .map(r => r.code match {
             // HTTP 404 is a "normal" error: we're trying to lookup future blocks that haven't been mined.
-            case StatusCodes.NotFound => Seq.empty
-            case _ => r.unsafeBody.arr
+            case StatusCode.NotFound => Seq.empty
+            case _ => r.body.fold(exc => throw exc, jvalue => jvalue).arr
```

Comment:
This pattern comes regularly enough that we may want to create a utility function that does this fold, don't we?
This could even be called `getBodyUnsafe` or something like that and very explicitly say that it throws?

Reply:
Indeed, besides a package rename, the main change was the removal of `unsafeBody` which I replaced by this `fold`. The idea was to just "make it work" now, and later improve our code to be more in line with their functional `scodec` style.

I could use the pimp pattern to define an `unsafeBody`, but it wouldn't make the diff much smaller, due to the other changes. If you think it's worth it, I can invest some time to make our code more functional and remove those `fold` alltogether.

Reply:
No, let's keep it like that for this PR, it's good to have the dependencies updated.
That can be done in a separate PR :+1: